test/unittest/tests/cursors/test_bulk_cursor.cpp

```diff
@@ -0,0 +1,379 @@
+{
+    session->checkpoint(session, nullptr);
+}
+
+/*
+ * thread_function_drop --
+ *     This function is designed to be used as a thread function, and force drops a table.
+ */
+static void
+thread_function_drop(WT_SESSION *session, std::string const &uri)
+{
+    session->drop(session, uri.c_str(), "force=true");
+}
+
+/*
+ * print_dhandles
+ *     For diagnostics of any failing tests, prints the dhandles on a session.
+ */
+static void
+print_dhandles(WT_SESSION_IMPL *session_impl)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DATA_HANDLE *dhandle;
+
+    printf("Session dhandle: %p\n", session_impl->dhandle);
+    conn = S2C(session_impl);
+
+    TAILQ_FOREACH (dhandle, &conn->dhqh, q) {
+        printf(".   dhandle 0c%p\n", dhandle);
+    }
+}
+
+/*
+ * check_txn_updates
+ *     For diagnostics of any failing tests, prints information about mod values in a txn.
+ */
+static bool
+check_txn_updates(std::string const &label, WT_SESSION_IMPL *session_impl, bool diagnostics)
+{
+    bool ok = true;
+
+    if (diagnostics) {
+        WT_TXN *txn = session_impl->txn;
+
+        printf("check_txn_updates() - %s\n", label.c_str());
+        print_dhandles(session_impl);
+        printf("  txn = 0x%p, txn->id = 0x%" PRIu64 ", txn->mod = 0x%p, txn->mod_count = %u\n", txn,
+          txn->id, txn->mod, txn->mod_count);
+
+        WT_TXN_OP *op = txn->mod;
+        for (u_int i = 0; i < txn->mod_count; i++, op++) {
+            switch (op->type) {
+            case WT_TXN_OP_NONE:
+            case WT_TXN_OP_REF_DELETE:
+            case WT_TXN_OP_TRUNCATE_COL:
+            case WT_TXN_OP_TRUNCATE_ROW:
+                break;
+            case WT_TXN_OP_BASIC_COL:
+            case WT_TXN_OP_BASIC_ROW:
+            case WT_TXN_OP_INMEM_COL:
+            case WT_TXN_OP_INMEM_ROW:
+                WT_UPDATE *upd = op->u.op_upd;
+                printf("    mod %u, op->type = %i, upd->txnid = 0x%" PRIx64 "\n", i, op->type,
+                  upd->txnid);
+                break;
+            }
+        }
+    }
+
+    return ok;
+}
+
+/*
+ * report_cache_status
+ *     For diagnostics of any failing tests, prints cache information.
+ */
+static void
+report_cache_status(WT_CACHE *cache, std::string const &label, bool diagnostics)
+{
+    if (diagnostics) {
+        printf("Cache (label is '%s'):\n", label.c_str());
+        printf(". pages_inmem:      %" PRIu64 "\n", cache->pages_inmem);
+        printf(". pages_evicted:    %" PRIu64 "\n", cache->pages_evicted);
+        printf(". bytes_image_intl: %" PRIu64 "\n", cache->bytes_image_intl);
+        printf(". bytes_image_leaf: %" PRIu64 "\n", cache->bytes_image_leaf);
+        printf(". pages_dirty_intl: %" PRIu64 "\n", cache->pages_dirty_intl);
+        printf(". pages_dirty_leaf: %" PRIu64 "\n", cache->pages_dirty_leaf);
+        printf(". bytes_dirty_intl: %" PRIu64 "\n", cache->bytes_dirty_intl);
+        printf(". bytes_dirty_leaf: %" PRIu64 "\n", cache->bytes_dirty_leaf);
+    }
+};
```

Comment:
The semicolon here surprises me. Is it necessary?

Reply:
The semicolon was unnecessary, and added by mistake. I've removed it.